tests/Doctrine/Tests_PHP74/Persistence/Reflection/TypedNoDefaultReflectionPropertyTest.php

```diff
@@ -0,0 +1,86 @@
+<?php
+
+namespace Doctrine\Tests_PHP74\Persistence\Reflection;
+
+use Doctrine\Persistence\Reflection\TypedNoDefaultReflectionProperty;
+use PHPUnit\Framework\TestCase;
+
+class TypedNoDefaultReflectionPropertyTest extends TestCase
+{
+    public function testGetValue() : void
+    {
+        $object = new TypedNoDefaultReflectionPropertyTestClass();
+
+        $reflProperty = new TypedNoDefaultReflectionProperty(TypedNoDefaultReflectionPropertyTestClass::class, 'test');
+
+        self::assertNull($reflProperty->getValue($object));
+
+        $object->test = 'testValue';
+
+        self::assertSame('testValue', $reflProperty->getValue($object));
+
+        unset($object->test);
+
+        self::assertNull($reflProperty->getValue($object));
+    }
+
+    public function testSetValueNull() : void
+    {
+        $reflection = new TypedNoDefaultReflectionProperty(TypedFoo::class, 'id');
+        $reflection->setAccessible(true);
+
+        $object = new TypedFoo();
+        $object->setId(1);
+
+        self::assertTrue($reflection->isInitialized($object));
+
+        $reflection->setValue($object, null);
+
+        self::assertNull($reflection->getValue($object));
+        self::assertFalse($reflection->isInitialized($object));
+    }
+
+    public function testSetValueNullOnNullableProperty() : void
+    {
+        $reflection = new TypedNoDefaultReflectionProperty(TypedNullableFoo::class, 'value');
+        $reflection->setAccessible(true);
+
+        $object = new TypedNullableFoo();
+
+        $reflection->setValue($object, null);
+
+        self::assertNull($reflection->getValue($object));
+        self::assertTrue($reflection->isInitialized($object));
```

Comment:
This assertion fails and I can't understand why… can anybody help?

Reply:
@sspat maybe?

Reply:
Ah I think I undestand what happened: I refactored the source class from my vendor dir, but copied the test from an outside dir since tests are not shipped in archives, and ended up with a version of the test that does not match the class.